web/src/Web.App/ViewModels/SchoolSpendingComparisonViewModel.cs

```diff
@@ -2,8 +2,97 @@
 
 namespace Web.App.ViewModels;
 
-public class SchoolSpendingComparisonViewModel(School school)
+public class SchoolSpendingComparisonViewModel
 {
-    public string? Name => school.SchoolName;
-    public string? Urn => school.URN;
+    private readonly RagRating[] _originalRating;
+    private readonly RagRating[] _customRating;
+
+    public SchoolSpendingComparisonViewModel(School school, RagRating[] originalRating, RagRating[] customRating)
+
+    {
+        Name = school.SchoolName;
+        Urn = school.URN;
+        _originalRating = originalRating;
+        _customRating = customRating;
+        GroupedComparisonResultsNoChange = GetGroupedComparisonResults(true);
+        GroupedComparisonResultsChange = GetGroupedComparisonResults(false);
+        HighHeadline = GetHeadlineResults("red");
+        MediumHeadline = GetHeadlineResults("amber");
+        LowHeadline = GetHeadlineResults("green");
+    }
+    public string? Name { get; }
+    public string? Urn { get; }
+    public (int OriginalCount, int CustomCount, string Change) HighHeadline { get; }
+    public (int OriginalCount, int CustomCount, string Change) MediumHeadline { get; }
+    public (int OriginalCount, int CustomCount, string Change) LowHeadline { get; }
+    public List<ComparisonResult> GroupedComparisonResultsNoChange { get; }
+    public List<ComparisonResult> GroupedComparisonResultsChange { get; }
+
+    private List<ComparisonResult> GetGroupedComparisonResults(bool match)
+    {
+        var results = new List<ComparisonResult>();
+
+        foreach (var originalItem in _originalRating)
+        {
+            var customItem = _customRating.FirstOrDefault(c => c.Category == originalItem.Category);
+
+            bool isMatch = originalItem.PriorityTag == customItem?.PriorityTag;
+
+            if ((match && isMatch) || (!match && !isMatch))
+            {
+                var result = new ComparisonResult
+                {
+                    OriginalPriorityTag = originalItem.PriorityTag,
+                    CustomPriorityTag = customItem?.PriorityTag,
+                    OriginalValue = originalItem.Value,
+                    CustomValue = customItem?.Value,
+                    OriginalPercentile = originalItem.Percentile,
+                    CustomPercentile = customItem?.Percentile,
+                    OriginalRAG = originalItem.RAG,
+                    Category = originalItem.Category,
+                };
+
+                results.Add(result);
+            }
+        }
+
+        return results.OrderBy(r => Lookups.StatusOrderMap[r.OriginalRAG ?? string.Empty])
+            .ThenBy(r => r.OriginalValue)
+            .ToList();
+    }
+
+    private (int OriginalCount, int CustomCount, string Change) GetHeadlineResults(string rag)
+    {
+        var originalCount = _originalRating.Where(c => c.RAG == rag).Count();
```

Comment:
chore:  `_originalRating.Count(c => c.RAG == rag)` saves a few bytes here and below